File: signprob.py
import numpy as np
import matplotlib.pyplot as plt

deno = lambda x, p: np.exp(1.0j*p*x)
nume = lambda x, p: x**2 * np.exp(1.0j*p*x)
obs = lambda x, p: np.abs(nume(x, p) / deno(x, p) / (1 - p**2))

Nsample = 10000
list_p = np.arange(0, 100, 2)
list_obs = [np.mean([obs(np.random.normal(0.0, 1.0, 1), p) for i in np.arange(Nsample)]) for p in list_p]
plt.plot(list_p, list_obs)
plt.yscale('log')
plt.show()

File: signprob.jl
using Plots, Statistics, Random, Distributions

deno(x, p) = exp(im*p*x)
nume(x, p) = x^2 * exp(im*p*x)
Obs(x, p) = abs(nume(x, p) / deno(x, p) / (1-p^2))

Nsample = 10000
list_p = 0:2:100
list_obs = [mean(Obs.(rand(Normal(), Nsample), p)) for p in list_p]

plot(list_p, list_obs, yaxis=:log, seriestype=:scatter)
